src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -119,6 +149,7 @@ public Collector newCollector() {
     @Override
     public ReduceableSearchResult reduce(Collection<Collector> collectors) {
         final List<HybridTopScoreDocCollector> hybridTopScoreDocCollectors = new ArrayList<>();
+        final List<HybridTopFieldDocSortCollector> hybridSortedTopDocCollectors = new ArrayList<>();
```

Comment:
getting these hybrid collectors can be in it's own method to keep `reduce` simple. Main question - do you see the scenario when both topdocs and sorting are possible? If not, then we can get away with 1 - simpler main method logic - either this or that, and 2 - maybe a single list of collectors, I hope that collector methods that we're calling here are common between all hybrid specific collectors

Reply:
Discussed offline. The refactoring will be catered once I will raise the bug fix Missing docs when concurrent search enabled in this branch.